### PR TITLE
Weather and Nav endpoints

### PR DESCRIPTION
**Pull request by stephenfabian**
# Description

Please check the type of change your PR introduces:
- [ ] Bugfix (non-breaking change which fixes an issue)
- [x] Feature (non-breaking change which adds functionality)
- [ ] Code style update (formatting, renaming)
- [ ] Refactoring (no functional changes, no api changes)
- [ ] Build related changes
- [ ] Documentation content changes
- [ ] Other (please describe):

 

Adds Weather and Nav Controller with index methods, updated routes, and index spec tests

 

### Type of change

Please check the relevant option.

- [x] Non-Breaking Change
- [ ] Breaking change (fix or feature that causes existing functionality to not work as expected)
- [ ] Big Breaking change (this breaks everything and I am sorry)

# How Has This Been Tested?

Please check the option(s) that best describe your tests.

- [x] New spec files were added
- [ ] Tests were added to existing spec files
- [ ] Some tests were changed
- [ ] No tests were changed
- [ ] All tests were changed (please explain what in God's name happened)

### Tests Description

- Please add further details of your tests here.

# Checklist:

- [x] I have performed a self-review of my code
- [ ] I have commented my code, particularly in hard-to-understand areas
- [x] I have added tests that prove my fix is effective or that my feature works
- [ ] I have removed all unused code or commented why I left it
- [ ] My changes generate no new warnings
- [x] New and existing tests pass locally with my changes

### CODE REVIEW

File: app/facades/lat_long_facade.rb [hunk 1]
```diff
@@ -4,6 +4,9 @@ class LatLongFacade < LatLongService
   def self.lat_long_object(address)
     data = address_to_coord(address)
     lat_long_hash = data[:data][0].slice(:latitude, :longitude)
+    lat_long_hash.each do |key, value|
+      lat_long_hash[key] = value.round(4)
+    end
```
Comment (SandyyMarie) [inline app/facades/lat_long_facade.rb:9]: Great add - thanks Stephen!

File: config/routes.rb [hunk 2]
```diff
@@ -3,6 +3,9 @@
 Rails.application.routes.draw do
   namespace :api do
     namespace :v1 do
+      # resources :nav, only: [:index]
+      get '/nav', to: 'nav#index'
```
Comment (SandyyMarie) [inline config/routes.rb:7]: might be fine to change back to resources for consistency now

File: app/controllers/api/v1/nav_controller.rb [hunk 3]
```diff
@@ -0,0 +1,14 @@
+# frozen_string_literal: true
+
+module Api
+  module V1
+    class NavController < ApplicationController
+      def index
+        addresses = JSON.parse(params.keys.first, symbolize_names: true)
+        render json: NavSerializer.new(NavFacade.nav_object(addresses[:start_address], addresses[:end_address]))
+        # JSON.parse(params.body, symbolize_names: true)
```
Comment (SandyyMarie) [inline app/controllers/api/v1/nav_controller.rb:9]: whoops sorry for leaving that in lol

**Review by SandyyMarie — APPROVED** [review]
Awesome! Glad we have these working.

**Review by eriggs0207 — APPROVED** [review]
This looks good.  I will try to hit the endpoint once I build the create stop action

**Pull request merged**